tests/build/build.go

```diff
@@ -170,29 +166,13 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build-ser
 				timeout = time.Second * 300
 				interval = time.Second * 1
 				Eventually(func() bool {
-					switch gitProvider {
-					case "github":
-						prs, err := f.AsKubeAdmin.CommonController.Github.ListPullRequests(helloWorldComponentGitSourceRepoName)
-						Expect(err).ShouldNot(HaveOccurred())
-
-						for _, pr := range prs {
-							if pr.Head.GetRef() == constants.PaCPullRequestBranchPrefix+customDefaultComponentName {
-								Expect(pr.GetBase().GetRef()).To(Equal(helloWorldComponentDefaultBranch))
-								return true
-							}
-						}
-						return false
-					case "gitlab":
-						mrs, err := f.AsKubeAdmin.CommonController.Gitlab.GetMergeRequests()
-						Expect(err).ShouldNot(HaveOccurred())
+					prs, err := gitClient.ListPullRequests(helloWorldRepository)
+					Expect(err).ShouldNot(HaveOccurred())
 
-						for _, mr := range mrs {
-							if mr.SourceBranch == constants.PaCPullRequestBranchPrefix+customDefaultComponentName {
-								Expect(mr.TargetBranch).To(Equal(helloWorldComponentDefaultBranch))
-								return true
-							}
+					for _, pr := range prs {
+						if pr.SourceBranch == constants.PaCPullRequestBranchPrefix+customDefaultComponentName {
```

Comment:
This expression presents many times. Could you please extract it into a separate function, probably its name could be `func constructPaCPRBranchName(componentName string)`. It's up to you to determine a name for it.